test/SemVer.unittest.js

```diff
@@ -0,0 +1,492 @@
+			"<1.2.3+12.34": "<1.2.3+12.34",
+			"1.2.3 - 3.2.1": ">=1.2.3 (<3.2.1 || =3.2.1)",
+			">3.4": ">=3.4 not(~3.4)",
+			"1 || 2 || 3": "^1 || ^2 || ^3",
+			"1.2.3 - 3.2.1 || >3 <=4 || 1":
+				">=1.2.3 (<3.2.1 || =3.2.1) || >=3 not(^3) (<4 || ^4) || ^1"
+		};
+		for (const key of Object.keys(cases)) {
+			const expected = cases[key];
+			it(`should ${key} stringify to ${expected}`, () => {
+				expect(rangeToString(parseRange(key))).toEqual(expected);
+			});
+		}
+	});
+
+	describe("satisfies", () => {
+		const cases = {
+			// table cases
+			">=1": [
+				"1",
+				"2",
+				"!1-beta",
+				"!2-beta",
+				"1.2",
+				"!1.2-beta",
+				"2.2",
+				"!2.2-beta",
+				"1.beta",
+				"!1.beta-beta",
+				"!2.beta-beta"
+			],
+			">=1-beta": [
+				"1",
+				"2",
+				"1-beta",
+				"!1-gamma",
+				"!1-alpha",
+				"!2-beta",
+				"1.2",
+				"!1.2-beta",
+				"2.2",
+				"!2.2-beta",
+				"1.beta",
+				"!1.beta-beta",
+				"2.beta",
+				"!2.beta-beta"
+			],
+			">=1.2": [
+				"!1",
+				"2",
+				"!1-beta",
+				"!2-beta",
+				"!1.1",
+				"1.2",
+				"1.3",
+				"2.1",
+				"2.2",
+				"2.3",
+				"!1.beta",
+				"2.beta"
+			],
+			"~1.2": [
+				"!1",
+				"!2",
+				"!10",
+				"!1-beta",
+				"!2-beta",
+				"!1.1",
+				"1.2",
+				"!1.3",
+				"!1.20"
+			],
+			">=1.beta": [
+				"!1",
+				"2",
+				"!1-beta",
+				"!2-beta",
+				"!1.2",
+				"2.2",
+				"!1.alpha",
+				"1.beta",
+				"!1.gamma",
+				"2.beta"
+			],
+			// fixed cases
+			"2": [
+				"2",
+				"2.0.0",
+				"2.99.99",
+				"!2.3.4-beta",
+				"!2.3.4-beta.1",
+				"!2.3.4-beta+123",
+				"2.3.4+123",
+				"!1",
+				"!1.2.3",
+				"!3",
+				"!3.4.5"
+			],
+			"1.2.3-beta.1.2+33.44": [
+				"1.2.3-beta.1.2+33.44",
+				"!1.2.3-beta.1.2+33.44.55",
+				"!1.2.3-beta.1.2.3+33.44",
+				"!1.2.3.4-beta.1.2+33.44",
+				"!1.2.3-beta.1.2+33",
+				"!1.2.3-beta.1.2",
+				"!1.2.3-beta",
+				"!1.2-beta.1.2+33.44",
+				"!1.2.3+33.44",
+				"!1.2.3",
+				"!1"
+			],
+			"1.2.3+33.44": [
+				"1.2.3+33.44",
+				"!1.2.4+33.44",
+				"!1.2.3+33.55",
+				"!1.2.3-beta+33.44",
+				"!1.2.3+33.44.55",
+				"!1.2.3+33",
+				"!1.2+33.44",
+				"!1.2.3.4+33.44",
+				"!1.2.3",
+				"!1.2.4",
+				"!1.3",
+				"!1",
+				"!2"
+			],
+			"1.2.3-beta.1.2": [
+				"1.2.3-beta.1.2",
+				"!1.2.3-beta.1.2+33",
+				"!1.2.3-beta.1.2.3",
+				"!1.2.3.4-beta.1.2",
+				"!1.2.3-beta",
+				"!1.2-beta.1.2",
+				"!1.2.3+33",
+				"!1.2.3",
+				"!1"
+			],
+			"^2.3.4": [
+				"2.3.4",
+				"2.3.5",
+				"2.4.0",
+				"!3.3.4",
+				"!1.5.6",
+				"!2.3.3",
+				"!2.3.4-beta",
+				"!2.3.5-beta",
+				"2.3.4.test",
+				"!2.3.test",
+				"2.3.4+33",
+				"2.3.5+33",
+				"2.4.0+33",
+				"2.3.4.5"
+			],
+			"~2.3.4": [
+				"2.3.4",
+				"2.3.5",
+				"!2.4.0",
+				"!3.3.4",
+				"!1.5.6",
+				"!2.3.3",
+				"!2.3.4-beta",
+				"!2.3.5-beta",
+				"2.3.4.test",
+				"!2.3.test",
+				"2.3.4+33",
+				"2.3.5+33",
+				"!2.4.0+33",
+				"2.3.4.5"
+			],
+			"!2.3": [
```

Comment:
Similar to the note above about `beta` and `gamma`, these should pass, but the `-rc` case is failing:

```
			"~1.2.0-beta": [
				"1.2.0-beta",
				"1.2.0-beta+1",
				"1.2.0-rc.0",
				"1.2.0",
				"1.2.1",
				"!1.2.0-alpha",
				"!1.2.1-beta",
				"!1.3.0-beta",
				"!1.3.0"
			],
```

Reply:
fixed, added the test case